sdk/src/main/java/io/dapr/client/domain/StateKeyValue.java

```diff
@@ -0,0 +1,56 @@
+package io.dapr.client.domain;
+
+public class StateKeyValue<T> {
+  private T value;
+  private String key;
+  private String etag;
+  private StateOptions options;
```

Comment:
We should not get this argument from user. If this is only for internal use, then there should be another class to represent this composition of KeyValue + StateOptions.

Also, make this class immutable. There should be no need to change it.

Reply:
That's there because it's needed like that for GRPC

Reply:
**Important**

This structure is the following:

1. For both gRPC and HTTP (mentioned before)
2. Is not internal. It's provided by the user.